gsa/src/web/pages/hosts/detailspage.js

```diff
@@ -263,93 +274,153 @@ Details.propTypes = {
   entity: PropTypes.model.isRequired,
 };
 
-const Page = ({
-  entity,
-  permissions = [],
-  onChanged,
-  onDownloaded,
-  onError,
-  onInteraction,
-  ...props
-}) => {
-  const goto_host = goto_details('host', props);
+const Page = () => {
+  // Page methods
+  const {id} = useParams();
+  const history = useHistory();
+  const [, renewSessionTimeout] = useUserSessionTimeout();
+  const [downloadRef, handleDownload] = useDownload();
+  const {
+    dialogState: notificationDialogState,
+    closeDialog: closeNotificationDialog,
+    showError,
+  } = useDialogNotification();
+
+  // Load host related entities
+  const {host, refetch: refetchHost, loading, error: entityError} = useGetHost(
+    id,
+  );
+  const {permissions = [], refetch: refetchPermissions} = useGetPermissions({
+    filterString: permissionsResourceFilter(id).toFilterString(),
+  });
+
+  // host related mutations
+  const exportEntity = useExportEntity();
+
+  const [deleteHost] = useDeleteHost();
+  const exportHost = useExportHostsByIds();
+
+  // host methods
+  const handleDeleteHost = deletedHost => {
+    return deleteHost(deletedHost.id)
+      .then(goto_list('hosts', {history}))
+      .catch(showError);
+  };
+
+  const handleDownloadHost = exportedHost => {
+    exportEntity({
+      entity: exportedHost,
+      exportFunc: exportHost,
+      resourceType: 'hosts',
+      onDownload: handleDownload,
+      showError,
+    });
+  };
+
+  // Timeout and reload
+  const timeoutFunc = useEntityReloadInterval(host);
+
+  const [startReload, stopReload, hasRunningTimer] = useReload(
+    refetchHost,
+    timeoutFunc,
+  );
+
+  useEffect(() => {
+    // start reloading if host is available and no timer is running yet
+    if (hasValue(host) && !hasRunningTimer) {
+      startReload();
+    }
+  }, [host, startReload]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  // stop reload on unmount
+  useEffect(() => stopReload, [stopReload]);
+
   return (
     <HostComponent
-      onTargetCreated={goto_entity_details('target', props)}
-      onTargetCreateError={onError}
-      onCreated={goto_host}
-      onDeleted={goto_list('hosts', props)}
-      onDownloaded={onDownloaded}
-      onDownloadError={onError}
-      onIdentifierDeleted={onChanged}
-      onIdentifierDeleteError={onError}
-      onInteraction={onInteraction}
-      onSaved={onChanged}
+      onTargetCreated={goto_entity_details('target', {history})}
```

Comment:
Shouldn't it be goto_details here too? I don't know the difference between both functions atm.

Reply:
goto_details is the old function and goto_entity_details is the new one for entities that use graphql mutations (which directly return the id for the new entity). So it is correct for onTargetCreated to use goto_entity_details because creating a target already works with graphql (because we needed it for the task dialog). Since the create mutation for hosts is not implemented in gsa yet we still need to use goto_details for onCreated for now.

Reply:
Thanks for the detailed explanation!